RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToClusterKernel.dev.cc

```diff
@@ -0,0 +1,810 @@
+        // However, only the first half of elements end up eventually used:
+        // hence, here, only wordCounter/2 elements are allocated.
+        auto fedId_d = cms::alpakatools::make_device_buffer<uint8_t[]>(queue, wordCounter / 2);
+        alpaka::memcpy(queue, word_d, wordFed.word(), wordCounter);
+        alpaka::memcpy(queue, fedId_d, wordFed.fedId(), wordCounter / 2);
+        // Launch rawToDigi kernel
+        alpaka::exec<Acc1D>(queue,
+                            workDiv,
+                            RawToDigi_kernel{},
+                            cablingMap,
+                            modToUnp,
+                            wordCounter,
+                            word_d.data(),
+                            fedId_d.data(),
+                            digis_d->view(),
+                            digiErrors_d->view(),
+                            useQualityInfo,
+                            includeErrors,
+                            debug);
+
+#ifdef GPU_DEBUG
+        alpaka::wait(queue);
+        std::cout << "RawToDigi_kernel was run smoothly!" << std::endl;
+#endif
+      }
+      // End of Raw2Digi and passing data for clustering
+
+      {
+        // clusterizer
+        using namespace pixelClustering;
+        // calibrations
+        using namespace calibPixel;
+        const int threadsPerBlockOrElementsPerThread = []() {
+          if constexpr (std::is_same_v<Device, alpaka_common::DevHost>) {
+            // NB: MPORTANT: This could be tuned to benefit from innermost loop.
+            return 32;
+          } else {
+            return 256;
+          }
+        }();
+        const auto blocks = cms::alpakatools::divide_up_by(std::max<int>(wordCounter, numberOfModules),
+                                                           threadsPerBlockOrElementsPerThread);
+        const auto workDiv = cms::alpakatools::make_workdiv<Acc1D>(blocks, threadsPerBlockOrElementsPerThread);
+
+        alpaka::exec<Acc1D>(
+            queue, workDiv, calibDigis{}, clusterThresholds, digis_d->view(), clusters_d->view(), gains, wordCounter);
+
+#ifdef GPU_DEBUG
+        alpaka::wait(queue);
+        std::cout << "countModules kernel launch with " << blocks << " blocks of " << threadsPerBlockOrElementsPerThread
+                  << " threadsPerBlockOrElementsPerThread\n";
+#endif
+
+        alpaka::exec<Acc1D>(
+            queue, workDiv, countModules<TrackerTraits>{}, digis_d->view(), clusters_d->view(), wordCounter);
+
+        auto moduleStartFirstElement =
+            cms::alpakatools::make_device_view(alpaka::getDev(queue), clusters_d->view().moduleStart(), 1u);
+        alpaka::memcpy(queue, nModules_Clusters_h, moduleStartFirstElement);
+        const auto threadsPerBlockFindClus =
+            512;  //((TrackerTraits::maxPixInModule / 16 + 128 - 1) / 128) * 128;  /// should be larger than maxPixInModule/16 aka (maxPixInModule/maxiter in the kernel)
```

Comment:
Is the commented out formula useful? Maybe move the comments on preceding line(s) for better formatting?